packages/babel-plugin-transform-typescript/src/index.js

```diff
@@ -205,11 +205,19 @@ export default function() {
       },
 
       TSImportEqualsDeclaration(path) {
-        throw path.buildCodeFrameError("`import =` is not supported.");
+        throw path.buildCodeFrameError(
+          "`import =` is not supported in Babel's Typescript parser " +
```

Comment:
I'd use `plugin` instead of `parser`, since it is parsed correctly but not transformed.

Reply:
It is not. Because If you don't use any plugins. It breaks still..

Reply:
Or maybe `is not supported by @babel/plugin-transform-typescript`? :thinking:

Reply:
> It is not. Because If you don't use any plugins. It breaks still..

 If you don' transform it, it works. https://runkit.com/embed/bsijdhpw0mq9

Reply:
Not sure either. I wish Babel could parse this to "importdeclaration = Identifier"
I could write codemod to migrate all our legacyJS files to TS. :)

Reply:
Yes, let's go with `is not supported by @babel/plugin-transform-typescript`.

Reply:
Sure.. On its way

Reply:
Just use `import identifier from "module"` :stuck_out_tongue: 

`import =` is only kept for backward compatibility, so it is "legacyTS" https://github.com/Microsoft/TypeScript/blob/master/doc/spec.md#1133-import-require-declarations

Reply:
Alright Just a grammar review here before I add
```
throw path.buildCodeFrameError(
          "`import =` is not supported by @babel/plugin-transform-typescript \n" +
            "Please consider using " +
            "`import <moduleName> from '<moduleName>';` alongside " +
            "Typescript's --allowSyntheticDefaultImports option.",
        );
```